Lib/turtledemo/clock.py

```diff
@@ -52,62 +51,54 @@ def clockface(radius):
             jump(-radius)
         rt(6)
 
+def display_date_time():
+    writer.clear()
+    now = datetime.now()
+    writer.home()
+    writer.forward(distance=65)
+    writer.write(now.strftime(format="%A"),
+                 align="center", font=("TkFixedFont", 14, "bold"))
+    writer.back(distance=150)
+    writer.write(now.strftime(format="%Y/%m/%d"),
+                 align="center", font=("TkFixedFont", 14, "bold"))
+    writer.forward(distance=85)
+
+def initialize_hand(shape, color):
+    hand = Turtle()
+    hand.shape(shape)
+    hand.color(*color)
+    hand.resizemode("user")
+    hand.shapesize(1, 1, 3)
+    hand.speed(0)
+    return hand
+
 def setup():
     global second_hand, minute_hand, hour_hand, writer
     mode("logo")
     make_hand_shape("second_hand", 125, 25)
-    make_hand_shape("minute_hand",  130, 25)
+    make_hand_shape("minute_hand",  115, 25)
     make_hand_shape("hour_hand", 90, 25)
     clockface(160)
-    second_hand = Turtle()
-    second_hand.shape("second_hand")
-    second_hand.color("gray20", "gray80")
-    minute_hand = Turtle()
-    minute_hand.shape("minute_hand")
-    minute_hand.color("blue1", "red1")
-    hour_hand = Turtle()
-    hour_hand.shape("hour_hand")
-    hour_hand.color("blue3", "red3")
-    for hand in second_hand, minute_hand, hour_hand:
-        hand.resizemode("user")
-        hand.shapesize(1, 1, 3)
-        hand.speed(0)
+
+    second_hand = initialize_hand("second_hand", ("gray20", "gray80"))
+    minute_hand = initialize_hand( "minute_hand", ("blue1", "red1"))
+    hour_hand = initialize_hand( "hour_hand", ("blue3", "red3"))
+
     ht()
     writer = Turtle()
-    #writer.mode("logo")
     writer.ht()
     writer.pu()
     writer.bk(85)
-
-def wochentag(t):
-    wochentag = ["Monday", "Tuesday", "Wednesday",
-        "Thursday", "Friday", "Saturday", "Sunday"]
-    return wochentag[t.weekday()]
-
-def datum(z):
-    monat = ["Jan.", "Feb.", "Mar.", "Apr.", "May", "June",
-             "July", "Aug.", "Sep.", "Oct.", "Nov.", "Dec."]
-    j = z.year
-    m = monat[z.month - 1]
-    t = z.day
-    return "%s %d %d" % (m, t, j)
+    display_date_time()
```

Comment:
Why is this here instead of in tick?  Update is rare, but not impossible.  Does the rewriting lock the text?

Reply:
https://github.com/user-attachments/assets/9353818c-9504-4009-94a2-f60b13f29bbd